system_tests/dump/text_section.ll

```diff
@@ -0,0 +1,24 @@
+; Test for dumping of a fragment's text section. When pstore is built inside LLVM,
+; we use the latter's disassembler library to produce nice disassembled output
+; rather than ASCII-encoded binary. To pass we need simply to dump a section of
+; type text.
+
+; %S = the test source directory
+; %T = the test output directory
+; %binaries = the directories containing the executable binaries
+; %s = source path (path to the file currently being run)
+; %t = temporary file name unique to the test
+
+; REQUIRES: is_inside_llvm
+; RUN: rm -f "%t.db"
+; RUN: env REPOFILE="%t.db" "%binaries/opt" -O0 -S %s -o %t
+; RUN: env REPOFILE="%t.db" "%binaries/llc" -filetype=obj %t
+; RUN: "%binaries/pstore-dump" --all-fragments "%t.db" | "%binaries/FileCheck" %s
+
```

Comment:
It looks like the test doesn't check the nice disassembled output. Should the test check the disassembled output?

Reply:
It's a good point and one that I did give some thought. My logic here is:

- I’m not trying to test the LLVM disassembler. I can rely on another tests to do that.
- I don’t want to introduce any coupling between the disassembler's output and the pstore tests. Future changes in the disassembler shouldn’t cause the test to fail.
- I don’t want the test to depend on the presence of particular instructions which future changes in LLVM might alter.
- The bug was a crash: simply checking that the tool doesn’t crash is a step in the right direction!

Reply:
Thanks for giving me the detailed reasons.

> 
>     * I’m not trying to test the LLVM disassembler. I can rely on another tests to do that.
> 
>     * I don’t want to introduce any coupling between the disassembler's output and the pstore tests. Future changes in the disassembler shouldn’t cause the test to fail.
> 
>     * I don’t want the test to depend on the presence of particular instructions which future changes in LLVM might alter.
>

Fully agree. Sorry, I did not write my question clearly in my previous comment. We don't want to check the dumped assembling instruction contents; we only want to check there is no assertion when dumping the text section as assembling instructions. By the way, how to dump test section contents as assembling instructions? I cannot find the command option in the help message.

>     * The bug was a crash: simply checking that the tool doesn’t crash is a step in the right direction!

Did the tool crash even without dumping the text section as assembling instructions? If the pstore-dump is always crash, the ` "%binaries/pstore-dump" --all-fragments "%t.db" | "%binaries/FileCheck" %s` check is fine.

Reply:
> how to dump test section contents as assembling instructions? I cannot find the command option in the help message.

Is your question: “How do I enable disassembly of text sections”? If so, there is no command-line switch. If pstore-dump is built inside LLVM then this is always enabled.

> Did the tool crash even without dumping the text section as assembling instructions?

If pstore is built inside LLVM, there is currently no option to disable the disassembler.

Reply:
Ok, I see. Thanks.